spec/operators/groupBy-spec.ts

```diff
@@ -1371,54 +1371,35 @@ describe('groupBy operator', () => {
     expectSubscriptions(e1.subscriptions).toBe(e1subs);
   });
 
-  it('should return inner that does not throw when faulty outer is unsubscribed early',
-  () => {
-    const values = {
-      a: '  foo',
-      b: ' FoO ',
-      d: 'foO ',
-      i: 'FOO ',
-      l: '    fOo    '
+  it('should not error for late subscribed inners if outer is unsubscribed before inners are subscribed', () => {
+    const source = hot('-----^----a----b-----a------b----a----b---#');
+    // Unsubscribe before the error happens.
+    const unsub =           '                   !';
+    // Used to hold two subjects we're going to use to subscribe to our groups
+    const subjects: Record<string, Subject<string>> = {
+      a: new Subject(),
+      b: new Subject()
     };
-    const e1 = hot('-1--2--^-a-b---d---------i-----l-#', values);
-    const unsub =         '      !';
-    const expectedSubs =  '^     !';
-    const expected =      '--g----';
-    const innerSub =      '                                ^';
-    const g =                                             '-';
-
-    const expectedGroups = {
-      g: TestScheduler.parseMarbles(g, values)
-    };
-
-    const innerSubscriptionFrame = TestScheduler
-      .parseMarblesAsSubscriptions(innerSub)
-      .subscribedFrame;
-
-    const source = e1.pipe(
-      groupBy(
-        (val: string) => val.toLowerCase().trim(),
-        (val: string) => val,
-        (group: any) => group.pipe(skip(7))
-      ),
-      map((group: any) => {
-        const arr: any[] = [];
-
-        rxTestScheduler.schedule(() => {
-          group.pipe(
-            phonyMarbelize()
-          ).subscribe((value: any) => {
-            arr.push(value);
-          });
-        }, innerSubscriptionFrame - rxTestScheduler.frame);
-
-        return arr;
```

Comment:
Note, this bit here caused the test to be faulty. It was "asynchronous" because of the scheduling above... but because of the faux-nature of the asynchronicity, was able to get an array values out here, which it should not have been able to do under "real scenarios".

I tested this with real code in a stackblitz and confirmed what the actual behavior was in 6.6.2 and again locally with the new code just to be sure. Then I refactored the test to be something that works a bit better, with comments.